rasa/configurations/autoconfig.py

```diff
@@ -0,0 +1,60 @@
+import logging
+
+from typing import Text, Dict, Any, List
+
+from rasa.constants import CONFIG_AUTOCONFIGURABLE_KEYS
+from rasa.utils import io as io_utils
+
+logger = logging.getLogger(__name__)
+
+
+# In future iterations, this function and the function `create_config_for_keys` below
+# will get additional input parameters to allow for smarter configuration. E.g.
+# training_data: TrainingData
+# domain: Domain
+# stories: StoryGraph
+def get_autoconfiguration(config):
+    autoconfig_keys = CONFIG_AUTOCONFIGURABLE_KEYS
+    missing_keys = [k for k in autoconfig_keys if k not in config]
+
+    create_config_for_keys(config, missing_keys)
+
+    return config
+
+
+def create_config_for_keys(config: Dict[Text, Any], keys: List[Text]) -> None:
+    import pkg_resources
+
+    if keys:
+        logger.debug(
+            f"The provided configuration does not contain the key(s) {keys}. "
+            f"Running automatic configuration for them now."
+        )
+
+    default_config_path = pkg_resources.resource_filename(
+        __name__, "default_config.yml"
+    )
+
+    default_config = io_utils.read_config_file(default_config_path)
+
+    if not config.get("autoconfigured"):
```

Comment:
why do we save that to a different config key `autoconfigured`?

Reply:
To remember which keys were automatically configured -> when dumping, we know that those are the ones to add/overwrite. Let's call this option A.
An alternative (option B) would be: when dumping, just check again which keys are not provided in the `config.yml`. That gives us the same information, unless the `config.yml` was changed in between and a key added, in which case option A would overwrite that key in the `config.yml` with the auto configuration, whereas option B would leave it as is.

So I think B is better then, I'll change it accordingly and remove the `auto configured` key.

Reply:
Sounds good 👍🏼Thanks for outlining both options!